test/use-checkout-pricing.test.js

```diff
@@ -1,308 +1,119 @@
+      const { result, waitForNextUpdate } = renderUseCheckoutPricing({
+        subscriptions: [{
+          plan: 'basic',
+          addons: [{ code: 'snarf', quantity: 5 }]
+        }]
       });
-    });
-  });
 
-  describe('setPricing update function', () => {
-    it('should call subscription pricing method when passed an object with subscriptions', async () => {
-      const initialInput = { subscriptions: [{ plan: 'basic', quantity: 2 }] };
-      const { result, waitForNextUpdate } = renderUseCheckoutPricing(initialInput);
+      await waitForNextUpdate();
 
-      await act(async () => {
-        await expect(subscriptionPricingReturn.plan).toHaveBeenCalledWith('basic', { quantity: 2 });
-        const setPricing = result.current[1];
-        setPricing({ subscriptions: [{ plan: 'plan', quantity: 5 }] });
-        await waitForNextUpdate();
-        await expect(subscriptionPricingReturn.plan).toHaveBeenCalledWith('plan', { quantity: 5 });
-      });
+      const [{ price }] = result.current;
+      expect(price.next.total).toBe('24.99');
     });
-  });
-
-  describe('Rest inputs', () => {
-    const checkoutPricingMethods = ['giftCard', 'address', 'shippingAddress', 'tax'];
-
-    checkoutPricingMethods.forEach(async checkoutPricingMethod => {
-      it(`should call checkoutPricing#${checkoutPricingMethod} with input value`, async () => {
-        const initialInput = { subscriptions: [{ plan: 'basic', quantity: 2 }], [checkoutPricingMethod]: 'Some value' };
 
-        await act(async() => {
-          await renderUseCheckoutPricing(initialInput);
-        })
-
-        expect(checkoutPricingReturn[checkoutPricingMethod]).toHaveBeenCalledWith('Some value');
+    it('when given a currency and a subscription, then an updated currency', async () => {
+      const { result, waitForNextUpdate } = renderUseCheckoutPricing({
+        currency: 'EUR',
+        subscriptions: [{
+          plan: 'multiple-currencies',
+          quantity: 1,
+        }]
       });
-    });
-  });
 
-  describe('Call order', () => {
-    it('should call checkoutPricing.subscriptions before checkoutPricing.adjustments', async () => {
-      const initialInput = {
-        subscriptions: [{ plan: 'basic' }],
-        adjustments: [{ itemCode: 'item-1', quantity: 2 }]
-      };
+      await waitForNextUpdate();
+
+      const [state, setPricing] = result.current;
+      expect(state.price.next.total).toBe('19.99');
 
-      await act(async () => {
-        await renderUseCheckoutPricing(initialInput);
+      act(() => {
+        setPricing(prev => ({ ...prev, currency: 'USD' }));
       });
 
-      expect(checkoutPricingReturn.subscription).toHaveBeenCalled();
-      expect(checkoutPricingReturn.adjustment).toHaveBeenCalled();
-      expect(checkoutPricingReturn.subscription)
-        .toHaveBeenCalledBefore(checkoutPricingReturn.adjustment);
+      await waitForNextUpdate();
+      expect(state.price.next.total).toBe('19.99');
```

Comment:
The value here doesn't change from the initial value because they're the same for both currencies on the test-server.

I tested this and made sure it was working as intended and outputs the correct total in both cases (it is). Maybe we can update the test server to include another currency with a different value to assert against.

Reply:
This expectation should probably be around the currency value